streamer/proxy_node.py

```diff
@@ -198,7 +199,7 @@ def do_DELETE(self) -> None:
 
   @abc.abstractmethod
   def handle_non_chunked(self, path: str, length: int,
-                         file: BinaryIO) -> None:
+                         file: Union[BinaryIO, BufferedIOBase]) -> None:
```

Comment:
i think shutting down mypy at https://github.com/joeyparrish/shaka-streamer/blob/63aae3f9c304ebded7bdd260cc3af92e0e26dd32/streamer/proxy_node.py#L256 would be simpler/clearer.

Reply:
I don't think that addresses the issue.  Here's an example of the error I'm getting from CI when I turn it back on:

```
The type checker found the following errors: 
streamer/proxy_node.py:161: error: Argument 3 to "handle_non_chunked" of "RequestHandlerBase" has incompatible type "BufferedIOBase"; expected "BinaryIO"  [arg-type]
Found 1 error in 1 file (checked 19 source files)
```

https://github.com/joeyparrish/shaka-streamer/actions/runs/11561541232/job/32180782250

Reply:
This is slightly complicated by the fact that my laptop seems to think it's BinaryIO, but the GitHub VMs call it BufferedIOBase.  The two seem to be unrelated in terms of inheritance.

Reply:
ah sorry i didn't test this. the `type: ignore` should be where the error is.

Reply:
https://github.com/mariocynicys/shaka-streamer/actions/runs/11562438825/job/32183606442#step:6:8, yup that works, should we use it instead?

looks like there are other errors though (in lists and dicts typing) with different python versions https://github.com/mariocynicys/shaka-streamer/actions/runs/11562438825/job/32183606011

Reply:
I'm going to update the python versions we support, set the minimum to 3.9 going forward.

I would rather not use "type: ignore" where I can avoid it, because "A|B" is better than "whatever".  If we make a typo later, I don't want to "type: ignore" to keep us from finding out until runtime.